### ISSUE TITLE
Iteration Plan for Jan-Feb 2020

### ISSUE THREAD

**Issue by scarlett2018** [microsoft/nni#1900]
**Release Manager**: @QuanluZhang 
**Feature freeze date**: 2020/1/23
**Code freeze date**: 2/10 4PM 
_this will be a 5-week sprint_ , _mid sprint break: 1/24-1/30 Chinese Spring Festival_
**End Game Issue 2/17-2/23**: is tracking in [Jan-Feb 2020 End Game Plan #2025](https://github.com/microsoft/nni/issues/2025)
**Target release date**: 2/24/2020

-----------------------------------------------
### Open discussion / retrospective from previous release
- [x] Improve endgame process, i.e. auto IT for release branch. Follow up: add this to the release manual @chicm-ms @leckie-chn 
- [x] Stay with feature freeze deadline, to avoid big PR merge's risk by the end of release
     - problem: NAS algo might cause too much time to get the results (to prove the algo quality); resolutions: we might think of silent/experiment release for core functions.
-------------------------------------
### Planned Item
- [x] Finalize the integration with PT @SparkSnail @ultmaster @scarlett2018 
   - [x] support merge user PT config file with NNI config file @SparkSnail (PR ready for review in internal repo @QuanluZhang @chicm-ms )
   - [ ] build pypi for internal version @SparkSnail (pending for the above PR)
- [x] Proxyless NAS left work - reproduce and report the results (2/3 finish check) @QuanluZhang #1863 [need review Chi]

**TensorFlow 2.0 support @liuzhe-lz  (PR eta. 2/~6~ 9)**
- ~[ ] tensorflow 2.0 (tf.keras) support for NNI NAS interface (#1737 )  (1.8 pr) @liuzhe-lz~
- [x] tensorflow 2.0 support for ~static~ dynamic graph model compression   (depend on #1737 ) @chicm-ms 
- [ ] tensorflow 2.0 support for static graph model compression (2d)  @liuzhe-lz 
- [ ] tensorflow 2.0 support for static graph NAS (~1.13~ ~1.14 draft pr~ 1.17 pr)  @liuzhe-lz 
- [ ] tensorflow 2.0 test and example  (code 2d) @liuzhe-lz 
- [ ] TF2 - upgrade to the new cuda version for our maintained docker image (#1748 , 1 day, first test internally) @liuzhe-lz 

- [x] fabric framework adoption @lvybriage (ETA 2/7)
- [x] cancel trials in UNKNOWN state (#1735 , pr submitted, ) @chicm-ms @lvybriage
- [x] cancel trials in UNKNOWN state WebUI (pr submitted ) @lvybriage
- [x] BUG - correctly support `NaN` in metric data, JSON compliant (#1589 , 1.14 pr, ready for review)@chicm-ms
- [x] add a parameter to cmd to support nni command as top cmd (1.13 pr, ready for review @chicm-ms ) @SparkSnail 
- [x] add pipeline for frameworkcontroller (1.17 PR1971 @chicm-ms @liuzhe-lz ) @SparkSnail @chicm-ms 
- [x] support complete PAI configuration file (1.17 pr @chicm-ms) @SparkSnail 
- [x] update pai doc @SparkSnail 

**Model compression**
- [x] Support DataParallel (#1915 , committed to Dev Branch) @Cjkkkk 
  - [x] update 4 quantizers, and ActivationAPoZ/Mean, Level 3 pruners and  @Cjkkkk [PR submitted, need a discussion with @QuanluZhang and @chicm-ms at 4pm 2/7 **to discuss defer or not**; need to merge dev to master (2/9);]
  - [x] update L1/2Filter, FPGM, AGP 4 pruners @chicm-ms 
  - [x] update Lottery Ticket 1 pruner @QuanluZhang 
  - [x] update UT @chicm-ms 
- [x] Model transform for real speed up, an alpha version (#1846 / #1927 , 1.14 design review, 1.22 draft pr) @QuanluZhang

**Neural architecture search (NAS)**
- [x] C-DARTS @ultmaster (1.17 pr by Hongyuan, ready for review)
- [x] tests for supported NAS algorithms (#1598 , @liuzhe-lz @QuanluZhang PR review) @ultmaster
- [x] tests for data parallel NAS algorithms (#1598, @liuzhe-lz @QuanluZhang PR review) @ultmaster @chicm-ms 
- [x] prepare GPU vm for pipeline (1.14) @chicm-ms
 - [x] fix security issue in win vm @chicm-ms 
    
**Documentation**
- [x] add readthedoc to build process (Friday pr, ready for review @chicm-ms ) @leckie-chn
- [x] improvement of document structure (ready for review) @QuanluZhang @leckie-chn @scarlett2018  (#1827 To be reviewed)
- [x] documentation for NAS (1.20 pr) @ultmaster 
- [ ] customized homepage for sphinx doc build (before code freeze) @leckie-chn @lvybriage

------------------Deferred------
- [ ] Integration with DLWS //@QuanluZhang is following up with Qixiang @Gerhut , need more time.   
- [ ] bug - clone fail @liuzhe-lz #1985 
- [ ] AutoNLP @QuanluZhang 
     - [x] search algo @ultmaster
     - [ ] retrain //yaming
 - [ ] prepare GPU vm for pipeline (1.14), migrate remote docker to the GPU vm @SparkSnail 
- [ ] write doc for IPC interface (#1741 )  @liuzhe-lz
